antiweb.py

```diff
@@ -2780,44 +2794,94 @@ def write(path, fname, output, token, warnings, index, index_rst, recursive, con
+
+#.. csv-table::
+#   :header: "Input File", "Output File"
+
+#   ``C:\antiweb\testing.cs``, *C:\\antiweb\\testing.rst*
+#   ``C:\antiweb\testing.rst``, *C:\\antiweb\\testing_docs.rst*
+           
+           
+#The output file name for the index file is only the file name itself: *"testing.rst"*.  
+#The recursive option indicates that the a relative path to the file should be used in the index file: *"reportdir\\testing.rst"*
+
 #@code
 
         docs = "_docs"
-
+        rst_suffix = ".rst"
+        
         if not output:
+            out_file = os.path.splitext(fname)[0]
+            if fname.endswith(rst_suffix):
+                out_file =  out_file + docs
+            out_file = out_file + rst_suffix
+                
+            out_file_name_index = os.path.split(out_file)[1]
             if recursive:
-                if fname.endswith(".rst"):
-                    out_file_name = os.path.splitext(fname)[0] + docs + ".rst"
-                else:
-                    out_file_name = os.path.splitext(fname)[0] + ".rst"
-                out_file = out_file_name
-                out_file_name = os.path.relpath(out_file_name, path)
-            else:
-                if fname.endswith(".rst"):
-                    out_file = os.path.splitext(fname)[0] + docs + ".rst"
-                else:
-                    out_file = os.path.splitext(fname)[0] + ".rst"
-                out_file_name = os.path.split(out_file)[1]
+                out_file_name_index = os.path.relpath(out_file_name_index, path)
 
 #@edoc
-#There is an output given, so it can also be either recursive or not recursive. With the additional output parameter the file path gets split up and put together so it can be processed by ''process_file''. There is also a differentiation between Linux, Windows and OS X (because of the different paths in each operating system)
+
+#If there is an output given, we have to distinguishe between the non-recursive and recursive option:
+
+#.. csv-table::
+#   :header: "Input File", "Output File"
+
+#   ``C:\antiweb\testing.cs -o report.rst``,*working_dir\\report.rst*
+#   ``C:\antiweb\testing.cs -o report``,*working_dir\\report\\testing.rst*
+#   ``C:\antiweb\testing.cs -o report\report.rst``,*C:\\antiweb\\report\\report.rst*
+#   ``C:\antiweb\testing.cs -o report\report.rst\``,*C:\\antiweb\\report\\report.rst\\testing.rst*
+#   ``C:\antiweb\testing.cs -o C:\antiweb\report.rst``,*C:\\antiweb\\report.rst*
+
+#When the recursive option is used in combination with the output option, the output parameter is treated as the documentation directory:
+
+#.. csv-table::
+#   :header: "Input File", "Output File"
+
+#   ``C:\antiweb\ -r -o report.rst``,*working_dir\\report.rst\\file.rst*
+#   ``C:\antiweb\ -r -o report``,*working_dir\\report\\file.rst*
+#   ``C:\antiweb\ -r -o C:\antiweb\report.rst``,*C:\\antiweb\\report.rst\\file.rst*
+
+#The output file name for the index file is the file name itself: *testing.rst*.  
```

Comment:
I don't understand this paragraph:
- Which 'filename' is used as the index file?
- The path separator part is unclear !resolve